src/Uno.UI/UI/Xaml/Media/VisualTreeHelper.Android.cs

```diff
@@ -31,10 +31,10 @@ internal static void SwapViews(UIElement oldView, UIElement newView)
 
 				if (unoViewGroup != null)
 				{
-					var newContentAsFrameworkElement = newView as IFrameworkElement;
+					var newContentAsFrameworkElement = newView as FrameworkElement;
 					if (newContentAsFrameworkElement != null)
 					{
-						newContentAsFrameworkElement.TemplatedParent = (unoViewGroup as IFrameworkElement)?.TemplatedParent;
+						newContentAsFrameworkElement.SetTemplatedParent((unoViewGroup as FrameworkElement)?.GetTemplatedParent());
 					}
```

Comment:
this part should be deleted
tp should not be set outside of template initialization/recycling
further native part shouldve never interact with it?